src/reflect/scala/reflect/internal/transform/Erasure.scala

```diff
@@ -148,9 +148,19 @@ trait Erasure {
         apply(atp)
       case ClassInfoType(parents, decls, clazz) =>
         ClassInfoType(
-          if (clazz == ObjectClass || isPrimitiveValueClass(clazz)) Nil
+          if (clazz == ObjectClass || isPrimitiveValueClass(clazz) || parents.isEmpty) Nil
```

Comment:
Funny, I tried a similar variation but backed out of it -- can't remember why.
Maybe it was the cleaner version not removing duplicate Object from erasedParents? Not sure how that could happen. Will give it another shot!